src/ingest-pipeline/airflow/dags/undo_upload_reorganization.py

```diff
@@ -0,0 +1,276 @@
+                        if not item.name.endswith("-metadata.tsv")
+                    ]
+                    if not items_to_move:
+                        print(f"Warning: Dataset path {dataset_path} exists but is empty")
+
+                    for item in items_to_move:
+                        dest_item = target_path / item.name
+
+                        if item.is_dir():
+                            if dest_item.exists():
+                                if item.name == "extras":
+                                    print(
+                                        f"Extras directory will not be moved. Data should still exist in original upload."
+                                    )
+                                    continue
+
+                                if dryrun:
+                                    print(f"DRYRUN: Would remove existing directory {dest_item}")
+                                else:
+                                    shutil.rmtree(dest_item)
+
+                            if dryrun:
+                                print(f"DRYRUN: Would move directory {item} -> {dest_item}")
+                            else:
+                                shutil.move(str(item), str(dest_item))
+                        else:
+                            if dest_item.exists():
+                                if dryrun:
+                                    print(f"DRYRUN: Would remove existing file {dest_item}")
+                                else:
+                                    dest_item.unlink()
+
+                            if dryrun:
+                                print(f"DRYRUN: Would move file {item} -> {dest_item}")
+                            else:
+                                shutil.move(str(item), str(dest_item))
+
+                    moved_count += 1
+                else:
+                    warning_prefix = "DRYRUN: " if dryrun else ""
+                    print(f"{warning_prefix}Warning: Dataset path {dataset_path} does not exist")
+            except Exception as e:
+                error_prefix = "DRYRUN: " if dryrun else ""
+                print(f"{error_prefix}Error moving data for UUID {uuid}: {e}")
+
+        mode_str = "DRYRUN: Would have moved" if dryrun else "Successfully moved"
+        print(f"{mode_str} data for {moved_count} out of {len(uuid_to_data_path)} datasets")
+        return uuid_to_data_path
+
+    @task()
+    def update_statuses(dataset_uuids, dag_run: DagRun):
+        # Set upload to New and datasets to Error
+        upload_uuid = dag_run.conf["upload_uuid"]
+        crypt_auth_tok = encrypt_tok(
+            airflow_conf.as_dict()["connections"]["APP_CLIENT_SECRET"]
+        ).decode()
+        try:
+            pythonop_set_dataset_state(
+                **{
+                    "dag_run": dag_run,
+                    "dataset_uuid_callable": lambda **kwargs: upload_uuid,
+                    "ds_state": "New",
```

Comment:
If the plan is to re-validate the upload after the rollback New is Ok, if not, maybe Submit should be the correct status.